### ISSUE TITLE
Error in rep(0, n * (k - 1)) : invalid 'times' argument

### ISSUE THREAD

**Issue by aron0093** [wmacnair/psupertime#11]
Input is a matrix not SCE object.  Data and code to reproduce the issue is below.

[mwe_psupertime.zip](https://github.com/wmacnair/psupertime/files/3750974/mwe_psupertime.zip)

```
> psupertime(x = as.matrix(data, rownames = TRUE), y = labels, sel_genes = 'all', smooth=TRUE, n_folds = 3)
converting y to a factor. label ordering used for training psupertime is:
3, 4, 5, 6
7319 genes have insufficient expression and will not be used as input to psupertime
processing data
    checking for zero SD genes
    denoising data
    scaling data
    processed data is 328 cells * 14039 genes
cross-validation training, 3 folds:
    fold 1
Error in rep(0, n * (k - 1)) : invalid 'times' argument

**Comment by aron0093** [comment]
I don't face this error anymore. it was solved by encoding the labels as categories, i.e.

`factor(labels)`

`labels` itself is a vector of numerical values. I would suggest that this could be implemented as part of the data processing within psupertime.

**Comment by wmacnair** [comment]
Ok. I thought I already had a check that the labels should be a factor, but maybe not... Thanks for raising it.

**Comment by wmacnair** [comment]
I think this is now fixed in commit `785fb20`. There was a logical error where I converted the `y` variable to a factor, but I hadn't actually returned the changed variable... Please check and let me know whether it works!
